src/privileged/trackers/framescript.js

```diff
@@ -47,6 +47,8 @@ addEventListener("DOMContentLoaded", function(e) {
   // early, but that's a tradeoff in simplicity vs. correctness we're
   // willing to take.
   content.window.addEventListener("load", () => {
+    const passwordFields = content.document.querySelectorAll("input[type='password']");
+    telemetryData.login_form_on_page = !!passwordFields.length;
```

Comment:
I wonder if this needs to move to the `unload` or `beforeunload` event. This correctly detected a password input on a page like https://bugzilla.mozilla.org/enter_bug.cgi where the login form is rendered at load time. But it failed to detect the password field at https://www.reddit.com/ where the login form is rendered dynamically.

Reply:
I agree with adding it to the `beforeunload` event, there seems to be no reason why not to. I tested on reddit, and noticed the login is loaded in an iframe - so I'm not sure what we should do in that case. Since all the data we are gathering is for the top frame only, and if errors are caused by logins then that error would be in the iframe, not the top level frame.

Reply:
Ah, good catch ... probably we don't need to chase down every DOM tree looking for password inputs. We'll just make that an asterisk of the analysis, unless @tanvihacks thinks we need to know every case.